File: Serene/Serene.Script/Northwind/Region/RegionGrid.cs

namespace Serene.Northwind
{
    using jQueryApi;
    using Serenity;

    [ColumnsKey("Northwind.Region"), IdProperty("RegionID"), NameProperty("RegionDescription")]
    [DialogType(typeof(RegionDialog)), LocalTextPrefix("Northwind.Region"), Service("Northwind/Region")]
    public class RegionGrid : EntityGrid<RegionRow>
    {
        public RegionGrid(jQueryObject container)
            : base(container)
        {
        }
    }
}

File: Serene/Serene.Core/Modules/Northwind/Region/RegionGrid.ts
namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    export class RegionGrid extends Serenity.EntityGrid<RegionRow, any> {
        protected getColumnsKey() { return "Northwind.Region"; }
        protected getDialogType() { return <any>RegionDialog; }
        protected getIdProperty() { return RegionRow.idProperty; }
        protected getLocalTextPrefix() { return RegionRow.localTextPrefix; }
        protected getService() { return RegionService.baseUrl; }

        constructor(container: JQuery) {
            super(container);
        }
    }
}